File: lv2/42842.py
'''
카펫
https://school.programmers.co.kr/learn/courses/30/lessons/42842
'''
def solution(brown, yellow):
  sumCnt = brown + yellow                     # 블록들의 합을 만들 수 있는 약수들의 곱 중에 정답이 있음
  answerIdx = 1                               # 최종 정답을 위한 변수
  if sumCnt ** 0.5 == int(sumCnt ** 0.5):     # 블록의 합이 제곱수면 해당 값이 정답이므로 예외처리
    return [int(sumCnt ** 0.5), int(sumCnt ** 0.5)]
  for i in range(2, int(sumCnt ** 0.5) + 1):          # 약수를 모두 검색할 필요 없이 절반만 확인하면 되므로 루트를 취함
    if sumCnt % i == 0:                               # 약수인 경우에
      if yellow == ((sumCnt // i) -2) * (i - 2):      # 해당 수가 yellow의 블록 수와 일치하는지 확인
        answerIdx = i
        break
  return [sumCnt // answerIdx, answerIdx]

File: lv2_js/42842.js
`
카펫
https://school.programmers.co.kr/learn/courses/30/lessons/42842
`
function solution(brown, yellow) {
  let yellowW = 0;
  let yellowH = 0;
  for (let i = 1; i <= parseInt(Math.sqrt(yellow)); i++) { // 노란색의 약수가 노란색 가로, 세로를 결정함
    if (yellow % i === 0) { // 약수인 경우
      yellowW = parseInt(yellow / i);
      yellowH = i;
      if ((yellowW + yellowH) * 2 + 4 === brown) break; // 가로 세로 두개씩과 모퉁이 4개를 합친 것이 갈색이 됨. 이게 일치한다면 정답임
    }
  }
  return [yellowW + 2, yellowH + 2];  // 노란색의 둘레가 갈색이므로 두개씩 더한 것이 카펫임
}